src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -321,25 +303,19 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
             }
 
             await ReloadLoadedPackagesAsync();
-
-            AfterLoadUnload(package);
+            await AfterLoadUnloadAsync(package);
         }
 
-        private void BeforeLoadUnload() {
+        private void BeforeLoadUnload(IRPackageViewModel package) {
             if (_selectedTab == SelectedTab.InstalledPackages || _selectedTab == SelectedTab.LoadedPackages) {
                 IsLoading = true;
             }
+            package.IsChanging = true;
         }
 
-        private void AfterLoadUnload(IRPackageViewModel package) {
+        private async Task AfterLoadUnloadAsync(IRPackageViewModel package) {
+            await ReplaceItemsAsync();
```

Comment:
We need to check current tab, otherwise, if user has switched from `LoadedPackages` to `AvailablePackages`, we will remove progress ring while available packages are loading.